### PR TITLE
Rebalance workloads before running the upgrades tests

### PR DESCRIPTION
**Pull request by dgrisonnet**
**tl;dr,** we need to rebalance HA workloads with PVCs manually before running the upgrade tests to prevent scheduling issues post-upgrade in CI

During the 4.8 timeframe, the monitoring team had an issue when transitioning workloads with PVCs from soft-affinity to hard-affinity. It turned out that persistent volumes may be bound to an availability zone, which means that you can't just move them around nodes. To solve that, we introduced the HighlyAvailableWorkloadIncorrectlySpread alert that was meant to detect workloads that can't be transitioned to hard-affinity. It comes with a [runbook](https://github.com/openshift/runbooks/blob/master/alerts/HighlyAvailableWorkloadIncorrectlySpread.md), that describes the steps a cluster-admin has to follow in order to fix their cluster. This alert is available starting from OCP 4.9 and 4.8.4.

The issue that we recently noticed in [bz-1949262](https://bugzilla.redhat.com/show_bug.cgi?id=1949262) from telemetry data is that we need to have a mechanism in place to make sure that cluster-admin fix their cluster before moving from 4.8 to 4.9 so that we can re-configure high-availability on Prometheus, Alertmanager and ThanosRuler at some point in 4.9. For that, we have CMO setting `Upgradeable=false` if it detects that the workloads aren't correctly spread. https://github.com/openshift/cluster-monitoring-operator/pull/1330 (a backport to 4.8.z is planned)

So the goal of this PR is to mimic the action of the cluster-admins that fix their cluster before the upgrade 4.9 upgrade. This would allow us to prevent any scheduling issue once we re-enable HA configurations for Prometheus, Alertmanager, and ThanosRuler.

/cc @simonpasquier @wking

### CODE REVIEW

File: ci-operator/step-registry/ipi/install/rebalance-workloads/ipi-install-rebalance-workloads-commands.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    echo "pods matching ${LABEL_SELECTOR} in namespace ${NS} are correctly spread"
+    return
+  fi
+
+  echo "need to rebalance pods matching ${LABEL_SELECTOR} in namespace ${NS}..."
+
+  oc adm cordon "${NODE}"
+
+  # Get all pods except one to keep it on the original node
+  PODS=$(oc get pods -n "${NS}" -l "${LABEL_SELECTOR}" -o jsonpath=\{'.items[*].metadata.name'\} | tr ' ' '\n' | head -n-1)
```
Comment (deads2k) [inline ci-operator/step-registry/ipi/install/rebalance-workloads/ipi-install-rebalance-workloads-commands.sh:25]: This doesn't look right.  What if you select a terminated pod or a pod that is in a permanent containercreating state?

**Comment by deads2k** [comment]
Why is this is a manual step?  Why shouldn't the 4.8 product have the balancing the way we want it?

**Comment by deads2k** [comment]
/hold

I'd like to understand why we're requiring a fairly intricate, but apparently mechanical change to made by a cluster-admin instead of writing code in our operator to manage this transition in 4.8.z.

**Comment by dgrisonnet** [comment]
Closing in favor of #21564 which uses the new annotation approach.
